### PR TITLE
Reducing TabBar templates reactivity for better performance

### PR DESCRIPTION
**Pull request by gmsecrieru**
Current implementation of RocketChat.TabBar templates (`views/app/tabBar`) rely too much on Session.get('openedRoom') reactivity, and this could turn into a few problems both on client and server. 

Giving the follow scenario:

1. Open 3 different rooms
2. Click "Files List" TabBar panel on one of them
3. Switch to another room

Current implementation: `master`
roomfiles - master

Proposed changes: `bugfix-room-tabbar-reactivity`
roomfiles - proposed

This is certainly affecting other templates besides `uploadedFilesList.coffee`, and the potential of severity of problem in server-side is directly proportional to [number of current clients * opened rooms], which could be critical on wide-used instances such as the Demo Server.

Can I have this reviewed by @RocketChat/owners so I can move forward and apply the same approach to the remaining Templates?

### CODE REVIEW

File: client/views/app/tabBar/uploadedFilesList.coffee [hunk 1]
```diff
@@ -23,11 +23,12 @@ Template.uploadedFilesList.events
     if $(e.currentTarget).siblings('.icon-picture').length
       e.preventDefault()
 
-
 Template.uploadedFilesList.onCreated ->
   instance = this
+  currRid = Template.currentData().rid
   this.autorun ->
```
Comment (sampaiodiego) [inline client/views/app/tabBar/uploadedFilesList.coffee]: I think this `autorun` is not needed anymore.

File: client/views/app/tabBar/uploadedFilesList.coffee [hunk 2]
```diff
@@ -23,11 +23,12 @@ Template.uploadedFilesList.events
...
-    instance.subscribe 'roomFiles', Session.get('openedRoom')
+    instance.subscribe 'roomFiles', currRid
 
 Template.uploadedFilesList.onRendered ->
+  console.log 'onRendered'
```
Comment (marceloschmidt) [inline client/views/app/tabBar/uploadedFilesList.coffee]: Please remove this log.

**Comment by gmsecrieru** [comment]
Based on code comments I think this is a "Go", so I will move forward. Thanks!

**Comment by engelgabriel** [comment]
@gmsecrieru are you going to remove the console.log ?

**Comment by gmsecrieru** [comment]
@engelgabriel Yep, there are more changes to come here!

**Comment by gmsecrieru** [comment]
Same scenario, now for Members List: 

`master`
memberslist - master

Proposed
memberslist-proposed

A bazillion less computations :)

**Pull request merged**